src/CognitoCustomMessageProcessor/CognitoCustomMessageFunction.cs

```diff
@@ -23,6 +24,17 @@ public class CognitoCustomMessageFunction : EventFunctionBase<CognitoCustomMessa
         // Needed to be able to run
         public async Task<JsonElement> FunctionHandler(CognitoCustomMessageEvent input, ILambdaContext context)
         {
+            // skip customization for MFA related messages
+            var mfaTriggers = new List<string> 
+            { 
+                "CustomMessage_VerifyUserAttribute",
+                "CustomMessage_Authentication"
+            };
+
+            if (mfaTriggers.Contains(input.TriggerSource))
+            {
+                return JsonDocument.Parse(JsonSerializer.Serialize(input)).RootElement.Clone();
```

Comment:
There is a lot going on in this line that is both memory and time inefficient.
Serialize -> Parse -> Clone.

The Clone is redundant, you already have the RootElement and you are not doing anything else with the JsonDocument, so it can be returned directly.

If this is .Net 6+ you can call `JsonSerializer.SerializeToElement(input)`.

https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.serializetoelement?view=net-6.0#system-text-json-jsonserializer-serializetoelement-1(-0-system-text-json-jsonserializeroptions)